### ISSUE TITLE
Docker transparent bridge

### ISSUE THREAD

**Issue by fansgit** [docker/docker#24418]
When I use docker networking, I usually miss one feature I would call `transparent bridge`. I've been enlightened by [Virtualbox bridged networking](https://www.virtualbox.org/manual/ch06.html#network_bridged), so the basic idea is excerpted as follows,

_With potential `transparent bridge`, Docker would use a `networking plugin` on your host system that filters data from your physical network adapter._ 

_This allows Docker to intercept data from the physical network and inject data into it, effectively creating a new network interface in software. When a container is using such a new software interface, it looks to the host system as though the container were physically connected to the interface using a network cable: the host can send data to the container through that interface and receive data from it._ 

_This means that you can set up routing or bridging between the container and the rest of your network._

For this to work, Docker needs a `networking plugin`  on the host system. Is this kind of plugin already existing or beneficial to be made?

**Comment by cpuguy83** [comment]
@fansgit You can already setup routing to the rest of the network with the built-in bridge driver (via static routes), so I'm not sure what you are asking for.

There is also macvlan and ipvlan drivers that give you actual IP's on the host network.

**Comment by mavenugo** [comment]
@fansgit have you tried `macvlan` (in master) and `ipvlan` (experimental) drivers ? I think these will fit your need. : https://github.com/docker/docker/blob/master/experimental/vlan-networks.md

**Comment by fansgit** [comment]
@cpuguy83 and @mavenugo , thanks!

Can the `Docker-Container` receive its ip address from the `DHCP-Server` on the networking setup below, when I use one of the drivers among built-in bridge`docker0`, experimental `macvlan` and `ipvlan`? Could you guide me if it can?
```
DHCP-Server------------(eth0)Docker-Host
                          |----------------Docker-Container(DHCP enabled)
```

**Comment by thaJeztah** [comment]
I think that would require a custom IPAM plugin; some information can be found here; https://github.com/docker/libnetwork/blob/master/docs/ipam.md and https://github.com/docker/go-plugins-helpers/tree/master/ipam

**Comment by ausarneteru** [comment]
yeah, beign able to connect multiple usb or even using a powerfull usb to connect to multiple networks would be nice, if there is a way with static routing i may need to learn that, is there an example static routing for this.

**Comment by ausarneteru** [comment]
i am using pfsense in virtualbox, transparent bridge, then i create vboxnets and run --network vboxnet1 or 2 as many as pfsense will run.

**Comment by dreh23** [comment]
I believe this is what you (and me) are looking for: https://gist.github.com/nerdalert/3d2b891d41e0fa8d688c
